core/execution/special_orders.go

```diff
@@ -268,57 +248,183 @@ func (m *Market) updateLPOrders(
 		}
 	}
 
-	// this is a list of order which a LP distressed
-	var (
-		distressedOrders  []*types.Order
-		distressedParties = map[string]struct{}{}
-	)
+	subFn := func(order *types.Order) {
+		if order.OriginalPrice == nil {
+			order.OriginalPrice = order.Price.Clone()
+			order.Price.Mul(order.Price, m.priceFactor)
+		}
+		// set the status to active again
+		order.Status = types.OrderStatusActive
+		m.matching.ReSubmitSpecialOrders(order)
+		partiesPos[order.Party] = m.position.RegisterOrder(ctx, order)
+		orderEvts = append(orderEvts, events.NewOrderEvent(ctx, order))
+	}
 
 	// now we iterate over all the orders which
 	// were initially cancelled, and remove them
 	// from the list if the liquidity engine instructed to
-	// cancel them
-	var cancelEvts []events.Event
+	// cancel them, but also the list of all new orders to be created
 	for _, order := range allOrders {
-		if _, ok := distressedParties[order.Party]; ok {
-			// party is distressed, not processing
-			continue
-		}
+		order.UpdatedAt = now
 
 		// these order were actually cancelled, just send the event
 		if _, ok := cancelIDs[order.ID]; ok {
-			// cancelEvts = append(cancelEvts, events.NewOrderEvent(ctx, order))
-			// these orders were submitted exactly the same before,
-			// so there's no reason we would not be able to submit
-			// let's panic if an issue happen
-		} else {
-			// set the status to active again
-			order.Status = types.OrderStatusActive
-			conf, _, err := m.submitValidatedOrder(ctx, order)
-			if conf == nil || err != nil {
-				distressedOrders = append(distressedOrders, order)
-				distressedParties[order.Party] = struct{}{}
-			} else if len(conf.Trades) > 0 {
-				m.log.Panic("submitting liquidity orders after a reprice should never trade",
-					logging.Order(*order))
+			order.Status = types.OrderStatusCancelled
+			orderEvts = append(orderEvts, events.NewOrderEvent(ctx, order))
+			continue
+		}
+
+		subFn(order)
+	}
+
+	for _, order := range submits {
+		order.UpdatedAt = now
+		subFn(order)
+	}
+
+	// send cancel events
+	m.broker.SendBatch(orderEvts)
+
+	// an ordered list of positions
+	var (
+		positions     = make([]events.MarketPosition, 0, len(partiesPos))
+		marginsBefore = map[string]*num.Uint{}
+		id            = m.GetID()
+		assetID, _    = m.mkt.GetAsset()
```

Comment:
we sure this won't err?

Reply:
yep, couldn't even instantiate a market without this, it's a bit of a annoying api as well a panic might somehow be better